src/devices/cardin.c

```diff
@@ -92,40 +102,56 @@ static int cardin_callback(bitbuffer_t *bitbuffer) {
 				dip[8]='+';
 		}
 
-		fprintf(stdout, "                 123456789\n");
-		fprintf(stdout, "dipswitch      = %s\n\n",dip);
-
-		fprintf(stdout, "                 -->ON\n");
-		fprintf(stdout, "right button   = ");
+		// Jumpers for the right button
 		if((bb[0][2] & 3) == 3) {
-			fprintf(stdout,                  "2 --o (this is right button)\n");
-			fprintf(stdout, "                 1 --o\n");
+			strcpy(rbuttonJ2, "--o (this is right button)");
+			strcpy(rbuttonJ1, "--o");
 		}
 		if((bb[0][2] & 9) == 9) {
-			fprintf(stdout,                  "2 --o (this is right button)\n");
-			fprintf(stdout, "                 1 o--\n");
+			strcpy(rbuttonJ2, "--o (this is right button)");
+			strcpy(rbuttonJ1, "o--");
 		}
 		if((bb[0][2] & 12) == 12) {
-			fprintf(stdout,                  "2 o-- (this is left button or two buttons on same channel)\n");
-			fprintf(stdout, "                 1 o--\n");
+			strcpy(rbuttonJ2, "o-- (this is left button or two buttons on same channel)");
```

Comment:
You are right, I didn't catch that. But doesn't it only ensure that at least two bits are set? I can't read decimal very well (with hex I can visualize the bit pattern easy). Maybe that test is intended to be like: (bb[0][2] & 0xf) == 3 ...? (i.e bb[0][2] = 15 matches all four conditions!)